mne/stats/regression.py

```diff
@@ -134,3 +137,202 @@ def _fit_lm(data, design_matrix, names):
         mlog10_p_val[predictor] = -np.log10(p_val[predictor])
 
     return beta, stderr, t_val, p_val, mlog10_p_val
+
+
+def linear_regression_raw(raw, events, event_id=None, tmin=-.1, tmax=1,
+                          covariates=None, reject=None, flat=None, tstep=1.,
+                          decim=1, picks=None, solver='pinv'):
+    """Estimate regression-based evoked potentials/fields by linear modelling
+    of the full M/EEG time course, including correction for overlapping
+    potentials and allowing for continuous/scalar predictors. Internally, this
+    constructs a predictor matrix X of size n_samples * (n_conds * window
+    length), solving the linear system Y = bX and returning b as evoked-like
+    time series split by condition.
+
+    References
+    ----------
+    Smith, N. J., & Kutas, M. (2015). Regression-based estimation of ERP
+    waveforms: II. Non-linear effects, overlap correction, and practical
+    considerations. Psychophysiology, 52(2), 169-189.
+
+    Parameters
+    ----------
+    raw : instance of Raw
+        A raw object. Note: be very careful about data that is not
+        downsampled, as the resulting matrices can be enormous and easily
+        overload your computer. Typically, 100 Hz sampling rate is
+        appropriate - or using the decim keyword (see below).
+    events : ndarray of int, shape(n_events, 3)
+        An array where the first column corresponds to samples in raw
+        and the last to integer codes in event_id.
+    event_id : dict
+        As in Epochs; a dictionary where the values may be integers or
+        iterables of integers, corresponding to the 3rd column of
+        events, and the keys are condition names.
+    tmin : float | dict
+        If float, gives the lower limit (in seconds) for the time window for
+        which all event types' effects are estimated. If a dict, can be used to
+        specify time windows for specific event types: keys correspond to keys
+        in event_id and/or covariates; for missing values, the default (-.1) is
+        used.
+    tmax : float | dict
+        If float, gives the upper limit (in seconds) for the time window for
+        which all event types' effects are estimated. If a dict, can be used to
+        specify time windows for specific event types: keys correspond to keys
+        in event_id and/or covariates; for missing values, the default (1.) is
+        used.
+    covariates : dict-like | None
```

Comment:
Is the description otherwise clear?

This is basically to hook in a dataframe with scalar predictors (e.g. word length).

(I think the API might need to be expanded later, to also allow for pd.Series-like objects under every key, with their own sample points-index. See e.g. @njsmith 's comments.)